clang/lib/ExtractAPI/API.cpp

```diff
@@ -54,6 +54,20 @@ RecordContext *APIRecord::castToRecordContext(const APIRecord *Record) {
   }
 }
 
+void RecordContext::stealRecordChain(RecordContext &Other) {
+  // If we don't have an empty chain append Other's chain into ours.
+  if (First)
+    Last->NextInContext = Other.First;
```

Comment:
This is the tiniest of nitpicks, but the fact that this only checks `First` but then accesses `Last` (which semantically makes sense but is semantically suspect) feels wrong to me. 😅

Reply:
Documented the invariant that `First` being non-null implies that `Last` is also non-null in an assert above.